app/components/Form/ProjectQuarterlyReportForm.tsx

```diff
@@ -0,0 +1,315 @@
+        },
+      },
+      debounceKey: formChange.id,
+      optimisticResponse: {
+        updateFormChange: {
+          formChange: {
+            id: formChange.id,
+            newFormData: newFormData,
+            changeStatus: formChange.changeStatus,
+          },
+        },
+      },
+    });
+  };
+
+  const [discardFormChange] = useDiscardFormChange(
+    projectRevision.projectQuarterlyReportFormChanges.__id
+  );
+
+  const deleteQuarterlyReport = (
+    formChangeId: string,
+    formChangeOperation: FormChangeOperation
+  ) => {
+    discardFormChange({
+      formChange: { id: formChangeId, operation: formChangeOperation },
+      onCompleted: () => {
+        delete formRefs.current[formChangeId];
+      },
+    });
+  };
+
+  const stageQuarterlyReportFormChanges = async () => {
+    const validationErrors = Object.keys(formRefs.current).reduce(
+      (agg, formId) => {
+        const formObject = formRefs.current[formId];
+        return [...agg, ...validateFormWithErrors(formObject)];
+      },
+      []
+    );
+
+    const completedPromises: Promise<void>[] = [];
+
+    projectRevision.projectQuarterlyReportFormChanges.edges.forEach(
+      ({ node }) => {
+        if (node.changeStatus === "pending") {
+          const promise = new Promise<void>((resolve, reject) => {
+            applyUpdateFormChangeMutation({
+              variables: {
+                input: {
+                  id: node.id,
+                  formChangePatch: {
+                    changeStatus: "staged",
+                  },
+                },
+              },
+              debounceKey: node.id,
+              onCompleted: () => {
+                resolve();
+              },
+              onError: reject,
+            });
+          });
+          completedPromises.push(promise);
+        }
+      }
+    );
+    try {
+      await Promise.all(completedPromises);
+
+      if (validationErrors.length === 0) props.onSubmit();
+    } catch (e) {
+      // the failing mutation will display an error message and send the error to sentry
+    }
+  };
+
+  const [sortedQuarterlyReports, nextQuarterlyReportIndex] = useMemo(() => {
+    const filteredReports =
+      projectRevision.projectQuarterlyReportFormChanges.edges
+        .map(({ node }) => node)
+        .filter((report) => report.operation !== "ARCHIVE");
+
+    filteredReports.sort(
+      (a, b) =>
+        a.newFormData.reportingRequirementIndex -
+        b.newFormData.reportingRequirementIndex
+    );
+    const nextIndex =
+      filteredReports.length > 0
+        ? filteredReports[filteredReports.length - 1].newFormData
+            .reportingRequirementIndex + 1
+        : 1;
+
+    return [filteredReports, nextIndex];
+  }, [projectRevision.projectQuarterlyReportFormChanges]);
+
+  return (
+    <div>
+      <header>
+        <h2>Quarterly Reports</h2>
+
+        <SavingIndicator isSaved={!isUpdating && !isAdding} />
+      </header>
+
+      <Grid cols={10} align="center">
```

Comment:
I'm not a fan of using a grid. Its only use in this file is to display the `h3` and "remove" button on the same row, which can be done easily, better, and with much less code with a simple flexbox